### DISCUSSION TITLE
prisma generate needs to define server URL before running

### DISCUSSION THREAD

**Discussion by a-wf** [prisma/prisma#9636] [Q&A] [answered]
Hi, 
Since I upgraded Prisma to v3, I cannot generate client without set URL env variable. Is it normal? 

```
npx prisma generate --schema prisma/mySchema.prisma & npx prisma migrate dev --preview-feature --schema prisma/mySchema.prisma
Done in 0.99s.
Prisma schema loaded from prisma/mySchema.prisma
Error: Get config: Schema Parsing P1012
error: Environment variable not found: DATABASE_URL.
  -->  schema.prisma:3
   | 
 2 |   provider = "postgresql"
 3 |   url      = env("DATABASE_URL")
   | 
Validation Error Count: 1
```

**Comment by ryands17** [comment, marked as answer]
@akdj 👋 
Works fine for me on the latest version _3.2.0_. Could you just run `prisma generate` and check? It seems the error is coming from `prisma migrate dev`.

**Reply by a-wf** [reply]
Hi @ryands17 , 
yes, indeed, I did a two in one command `prisma generate & prisma migrate dev` and that make it fails. 
i just removed the migration part and it works fine. Thanks!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
